pkg/cmd/init.go

```diff
@@ -0,0 +1,435 @@
+			return fail.Runtime(err, "copying terraform configuration")
+		}
+
+		tfvars, err := os.Create(filepath.Join(opts.Path, "terraform.tfvars"))
+		if err != nil {
+			return err
+		}
+		defer tfvars.Close()
+
+		fmt.Fprintf(tfvars, "cluster_name=%q\n", opts.ClusterName)
+
+		for _, param := range prov.requiredTFVars {
+			fmt.Fprintf(tfvars, "%s\n", param)
+		}
+	}
+
+	return nil
+}
+
+var (
+	manifestTemplateSource = heredoc.Doc(`
+		apiVersion: {{ .APIVersion }}
+		kind: {{ .Kind }}
+		name: {{ .Name }}
+
+		cloudProvider:
+		  {{ .CloudProvider.Name }}: {}
+		{{- with .CloudProvider.External }}
+		  external: true
+		{{ end -}}
+		{{- with .CloudProvider.CloudConfig }}
+		  cloudConfig: |
+		{{ . | indent 4 -}}
+		{{ end -}}
+		{{- with .CloudProvider.CSIConfig }}
+		  csiConfig: |
+		{{ . | indent 4 -}}
+		{{ end }}
+		containerRuntime:
+		  containerd: {}
+
+		versions:
+		  kubernetes: {{ .Versions.Kubernetes }}
+		{{ with .MachineController }}
+		machineController:
+		  deploy: false
+		{{ end -}}
+
+		{{- with .OperatingSystemManager }}
+		operatingSystemManager:
+		  deploy: false
+		{{ end }}
+
+		{{- with .Addons }}
+		addons:
+		  enable: true
+		  addons:
+		{{- range .Addons }}
+		    - name: {{ .Name }}
+		{{ end }}
+		{{- end -}}
+	`)
+
+	manifestTemplate = template.Must(
+		template.New("manifest").Funcs(sprig.TxtFuncMap()).
+			Parse(manifestTemplateSource),
+	)
+)
+
+type genKubeOneClusterYAMLParams struct {
+	providerName      string
+	clusterName       string
+	kubernetesVersion string
+	validProviders    map[string]initProvider
+}
+
+func genKubeOneClusterYAML(params *genKubeOneClusterYAMLParams) ([]byte, error) {
+	prov := validProviders[params.providerName]
+
+	cluster := kubeonev1beta2.KubeOneCluster{
+		TypeMeta: kubeonev1beta2.NewKubeOneCluster().TypeMeta,
+		Name:     params.clusterName,
+		CloudProvider: kubeonev1beta2.CloudProviderSpec{
+			External:    prov.external,
+			CloudConfig: prov.cloudConfig,
+			CSIConfig:   prov.csiConfig,
+		},
+		ContainerRuntime: kubeonev1beta2.ContainerRuntimeConfig{
+			Containerd: &kubeonev1beta2.ContainerRuntimeContainerd{},
+		},
+		Versions: kubeonev1beta2.VersionConfig{
+			Kubernetes: params.kubernetesVersion,
+		},
+		Addons: &kubeonev1beta2.Addons{
+			Enable: true,
+			Addons: []kubeonev1beta2.Addon{
+				{
+					Name: "default-storage-class",
+				},
+			},
+		},
+	}
+
+	providerName := prov.alternativeName
+	if providerName == "" {
+		providerName = params.providerName
+	}
+
+	err := kubeonev1beta2.SetCloudProvider(&cluster.CloudProvider, providerName)
+	if err != nil {
+		return nil, err
+	}
+
+	if cluster.CloudProvider.None != nil {
+		cluster.Addons = nil
+		cluster.MachineController = &kubeonev1beta2.MachineControllerConfig{
+			Deploy: false,
+		}
+		cluster.OperatingSystemManager = &kubeonev1beta2.OperatingSystemManagerConfig{
+			Deploy: false,
+		}
+	}
+
+	var buf bytes.Buffer
+	err = manifestTemplate.Execute(&buf, &cluster)
```

Comment:
We could try to strict unmarshal the generated manifest to make sure that it's valid. We already do that in `kubeone config` command: https://github.com/kubermatic/kubeone/blob/1081e5f16e3d15e88b3a5a024c82a8ec13e1ab14/pkg/cmd/config.go#L280-L284